src/dbnode/storage/index/block.go

```diff
@@ -641,15 +641,16 @@ func (b *block) aggregateWithSpan(
 	}
 
 	var (
-		source     = opts.Source
-		size       = results.Size()
-		docsCount  = results.TotalDocsCount()
-		batch      = b.opts.AggregateResultsEntryArrayPool().Get()
-		batchSize  = cap(batch)
-		iterClosed = false // tracking whether we need to free the iterator at the end.
+		source      = opts.Source
+		size        = results.Size()
+		resultCount = results.TotalDocsCount()
+		batch       = AggregateResultsEntries(b.opts.AggregateResultsEntryArrayPool().Get())
+		maxBatch    = cap(batch)
+		iterClosed  = false // tracking whether we need to free the iterator at the end.
+		exhaustive  = false
```

Comment:
may not need this

Reply:
yeah seems like `opts.exhaustive(size, resultCount)` accomplishes the same goal